### DISCUSSION TITLE
Not able to create RTC connection when deployed to GCP with docker image

### DISCUSSION THREAD

**Discussion by pettersp** [whitphx/streamlit-webrtc#1070] [Q&A]
Hi,

I've created a simple streamlit view with a streamlit-webrtc component with the following configuration:

```
def video_frame_callback(frame: av.VideoFrame) -> av.VideoFrame:
    image = frame.to_ndarray(format="bgr24")

    annotated_frame, result = detection.detect_barcode(image)
    if not result:
        return av.VideoFrame.from_ndarray(image, format="bgr24")
    else:
        add_product(result)
        return av.VideoFrame.from_ndarray(annotated_frame, format="bgr24")

webrtc_streamer(
        key="object-detection",
        mode=WebRtcMode.SENDRECV,
        rtc_configuration={"iceServers": [{"urls": ["stun:stun.l.google.com:19302"]}]},
        video_frame_callback=video_frame_callback,
        media_stream_constraints={"video": True, "audio": False},
        async_processing=False
```

When I run it locally, and with a local docker container it works fine. But when I deploy it to GCP app engine it struggles with setting up SDP/RTC connection:

![Screenshot 2022-09-22 at 10 18 07](https://user-images.githubusercontent.com/92877028/191695330-abdcd850-d1df-4b3e-9c73-7edd9841afff.png)

I've tried several different STUN and TURN servers and I've opened communication for ingress and egress on all ports in GCP. 

Any ideas for what I might try next?

**Comment by whitphx** [comment]
> I've tried several different STUN and TURN servers and I've opened communication for ingress and egress on all ports in GCP.

hmm, sorry, I have not seen any solutions other than these...
